nan.h

```diff
@@ -1527,26 +1755,28 @@ class NanObjectWrap : public node::ObjectWrap {
 template <typename T, typename P>
 class NanWeakCallbackData {
  public:  // constructors
+  inline ~NanWeakCallbackData();
+
+ private:  // constructors
+  NAN_DISALLOW_ASSIGN_COPY_MOVE(NanWeakCallbackData)
+
+ public:
   typedef void (*Callback)(
       NanWeakCallbackData & data  // NOLINT(runtime/references)
       );
-  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
-    : parameter(param), callback(cb) {
-    NanAssignPersistent(persistent, handle);
-    Revive();
+  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback cb)
+    : persistent(), parameter(param), callback(cb) {
+    std::memcpy(&persistent, handle, sizeof(persistent));
```

Comment:
To verify you need to store the `v8::Persistent<T> *handle` that's passed to `NanWeakCallbackData()`? IIRC the pointer to a `v8::Persistent` will never change in its lifetime. So couldn't you do something like:
```cpp
NanWeakCallbackData(v8::Persistent<T>* handle, P* pram, Callback cb)
    : persistent(&handle),
      parameter(param),
      callback(cb) {
}
```
So then this would change to use a pointer:
```cpp
 private:  // data members
  v8::Persistent<T>* persistent;
```
You can then continue to do operations like the following:
```cpp
  bool IsNearDeath() const { return (*persistent).IsNearDeath(); }
```
Though I could also be completely missing something.